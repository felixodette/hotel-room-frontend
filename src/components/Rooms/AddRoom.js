import React, { useState } from 'react';

function AddRoom() {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [image, setImage] = useState('');
  const [size, setSize] = useState('');
  const [view, setView] = useState('');
  const [bedding, setBedding] = useState('');

  const submitRoom = (e) => {
    e.preventDefault();
    const room = {
      name,
      description,
      image,
      size,
      view,
      bedding,
    };

    const requestOptions = {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(room),
    };
    fetch('http://localhost:3000/api/v1/rooms/', requestOptions)
      .then((response) => response.json());

    setName('');
    setDescription('');
    setImage('');
    setSize('');
    setView('');
    setBedding('');
  };
  return (
    <div>
      <h2>Add Room</h2>
      <form
        onSubmit={submitRoom}
      >
        <div>
          <label htmlFor="name">
            Name:
            <input
              id="name"
              type="text"
              placeholder="Name"
              value={name}
              onChange={(e) => setName(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="description">
            Room Description:
            <input
              id="description"
              type="text"
              placeholder="Description"
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="image">
            Room Image:
            <input
              id="image"
              type="text"
              placeholder="Image URL"
              value={image}
              onChange={(e) => setImage(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="size">
            Room Size:
            <input
              id="size"
              type="text"
              placeholder="In squaremeter"
              value={size}
              onChange={(e) => setSize(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="view">
            Room View:
            <input
              id="view"
              type="text"
              placeholder="Sea, ocean.."
              value={view}
              onChange={(e) => setView(e.target.value)}
              required
            />
          </label>

        </div>
        <div>
          <label htmlFor="bedding">
            Bedding Option:
            <input
              id="bedding"
              type="text"
              placeholder="Double, triple, twin..."
              value={bedding}
              onChange={(e) => setBedding(e.target.value)}
              required
            />
          </label>

        </div>
        <button type="submit">
          Add
        </button>
      </form>
    </div>
  );

}
  export default AddRoom;

