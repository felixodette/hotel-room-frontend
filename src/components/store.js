import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers/index';

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
/* eslint-disable */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
/* eslint-enable */

export default store;
