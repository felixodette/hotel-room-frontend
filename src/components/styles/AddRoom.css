#add-room-container {
  min-height: 100%;
  background: linear-gradient(0deg, rgba(151,191,17, 0.8), rgba(151,191,17, 0.8)), url("https://cache.marriott.com/content/dam/marriott-renditions/BJVEB/bjveb-suite-0236-hor-wide.jpg?output-quality=70&interpolation=progressive-bilinear&downsize=1336px:*");
  background-size: cover;
  height: 100vh;
}
#add-room-hr {
  width: 50%;
  border-width: 2px;
  border-color:  white;
}
.bg-transparent-add-room{
  background-color:  rgba(151,191,17, 0.9);
}
#name, #description, #image, #size, #view, #bedding {
border: 2px solid white;
padding-left: 15px;
height: 35px;}

