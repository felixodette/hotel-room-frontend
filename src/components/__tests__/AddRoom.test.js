import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import AddRoom from '../Rooms/AddRoom';
import store from '../../redux/configureStore';

describe('AddRoom', () => {
  render(
    <Provider store={store}>
      <AddRoom />
    </Provider>,
  );

  test('Test the dom for success rendering elements', () => {
    const title = screen.getByPlaceholderText('Description');
    expect(title).toBeInTheDocument();
  });

  test('Test the dom for non existing element on the page', () => {
    const title = screen.queryByText('example');
    expect(title).not.toBeInTheDocument();
  });

  test('It renders the Rooms component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <AddRoom />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('It renders the Rooms component correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AddRoom />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
