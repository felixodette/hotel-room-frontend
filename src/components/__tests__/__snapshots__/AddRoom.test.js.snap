// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRoom It renders the Rooms component correctly 1`] = `
<div>
  <div
    class="container-fluid d-flex flex-column align-items-center h-100 mb-5"
    id="add-room-container"
  >
    <h2
      class="text-center  mt-5 fw-bold fs-1 text-white text-uppercase"
      id="add-room-heading"
    >
      Add Room
    </h2>
    <hr
      id="add-room-hr"
    />
    <p
      class="text-center  fs-6 text-white"
    >
      Would you like to sell your hotel room to our website?
       
      <br />
      No need to wait. Fill the forms below your room will be added to our website accordingly!
       
      <br />
      And start earn some money today!
    </p>
    <form
      class="col-md-6 d-flex align-items-center flex-column justify-content-center"
      id="add-room-form"
    >
      <label
        class="text-white  col-md-8 mb-3"
        for="name"
      >
        Room Name:
         
        <br />
        <input
          class="col-12 bg-transparent-add-room rounded"
          id="name"
          placeholder="Deluxe Double Room"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="text-white  col-md-8 mb-3"
        for="description"
      >
        Room Description:
         
        <br />
        <input
          class="col-12 bg-transparent-add-room rounded"
          id="description"
          placeholder="Description"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="text-white  col-md-8 mb-3"
        for="image"
      >
        Room Image:
         
        <br />
        <input
          class="col-12 bg-transparent-add-room rounded"
          id="image"
          placeholder="Image URL"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="text-white  col-md-8 mb-3"
        for="size"
      >
        Room Size:
         
        <br />
        <input
          class="col-12 bg-transparent-add-room rounded"
          id="size"
          placeholder="In square meter"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="text-white  col-md-8 mb-3"
        for="view"
      >
        Room View:
         
        <br />
        <input
          class="col-12 bg-transparent-add-room rounded"
          id="view"
          placeholder="Sea, ocean.."
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="text-white  col-md-8 mb-3"
        for="bedding"
      >
        Bedding Option:
         
        <br />
        <input
          class="col-12 bg-transparent-add-room rounded"
          id="bedding"
          placeholder="Double, triple, twin..."
          required=""
          type="text"
          value=""
        />
      </label>
      <button
        class="fw-bold"
        id="add-room-submit-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
    <span />
  </div>
</div>
`;

exports[`AddRoom It renders the Rooms component correctly 2`] = `
<div
  className="container-fluid d-flex flex-column align-items-center h-100 mb-5"
  id="add-room-container"
>
  <h2
    className="text-center  mt-5 fw-bold fs-1 text-white text-uppercase"
    id="add-room-heading"
  >
    Add Room
  </h2>
  <hr
    id="add-room-hr"
  />
  <p
    className="text-center  fs-6 text-white"
  >
    Would you like to sell your hotel room to our website?
     
    <br />
    No need to wait. Fill the forms below your room will be added to our website accordingly!
     
    <br />
    And start earn some money today!
  </p>
  <form
    className="col-md-6 d-flex align-items-center flex-column justify-content-center"
    id="add-room-form"
    onSubmit={[Function]}
  >
    <label
      className="text-white  col-md-8 mb-3"
      htmlFor="name"
    >
      Room Name:
       
      <br />
      <input
        className="col-12 bg-transparent-add-room rounded"
        id="name"
        onChange={[Function]}
        placeholder="Deluxe Double Room"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="text-white  col-md-8 mb-3"
      htmlFor="description"
    >
      Room Description:
       
      <br />
      <input
        className="col-12 bg-transparent-add-room rounded"
        id="description"
        onChange={[Function]}
        placeholder="Description"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="text-white  col-md-8 mb-3"
      htmlFor="image"
    >
      Room Image:
       
      <br />
      <input
        className="col-12 bg-transparent-add-room rounded"
        id="image"
        onChange={[Function]}
        placeholder="Image URL"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="text-white  col-md-8 mb-3"
      htmlFor="size"
    >
      Room Size:
       
      <br />
      <input
        className="col-12 bg-transparent-add-room rounded"
        id="size"
        onChange={[Function]}
        placeholder="In square meter"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="text-white  col-md-8 mb-3"
      htmlFor="view"
    >
      Room View:
       
      <br />
      <input
        className="col-12 bg-transparent-add-room rounded"
        id="view"
        onChange={[Function]}
        placeholder="Sea, ocean.."
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="text-white  col-md-8 mb-3"
      htmlFor="bedding"
    >
      Bedding Option:
       
      <br />
      <input
        className="col-12 bg-transparent-add-room rounded"
        id="bedding"
        onChange={[Function]}
        placeholder="Double, triple, twin..."
        required={true}
        type="text"
        value=""
      />
    </label>
    <button
      className="fw-bold"
      id="add-room-submit-btn"
      type="submit"
    >
      Submit
    </button>
  </form>
  <span />
</div>
`;
